lib/src/client/options.dart

```diff
@@ -50,11 +51,17 @@ class ChannelOptions {
   final BackoffStrategy backoffStrategy;
   final String userAgent;
 
+  /// The maximum allowed time to wait for a connection to be established.
+  /// If [connectTimeout] is longer than the system level timeout duration,
+  /// a timeout may occur sooner than specified in [connectTimeout].
+  final Duration connectTimeout;
+
   const ChannelOptions({
     this.credentials = const ChannelCredentials.secure(),
     this.idleTimeout = defaultIdleTimeout,
     this.userAgent = defaultUserAgent,
     this.backoffStrategy = defaultBackoffStrategy,
     this.connectionTimeout = defaultConnectionTimeOut,
+    this.connectTimeout = defaultConnectTimeout
```

Comment:
If we set `connectTimeout` to 2 minutes now then we should definitely have a CHANGELOG.md entry for this, because it is a change in behavior. 

Alternatively change default behavior to the old behavior (infinite timeout - which seems to match what other implementations are doing)